tokio/src/sync/notify.rs

```diff
@@ -520,4 +531,25 @@ impl Drop for Notified<'_> {
     }
 }
 
+/// # Safety
+///
+/// `Waiter` is forced to be !Unpin.
+unsafe impl linked_list::Link for Waiter {
+    type Handle = *mut Waiter;
```

Comment:
it looks like, because this is an associated type, it cannot be `Pin<&mut Waiter>` because the lifetime of the `&mut` would be unbound...that's a bummer. i wonder if there's a way to work around that...

Reply:
It is just an annotation. You can implement it for `Marker<'a>`

Reply:
https://github.com/tokio-rs/tokio/blob/9abfc7e8005af2b7b3f7193fcedcf9e6ddcc6654/tokio/src/util/linked_list.rs#L191